### ISSUE TITLE
Sprint 4: Signup/Login System

### ISSUE THREAD

**Issue by Forsm0** [Forsm0/SWD-DineSmart#31]
**Database Setup**

- [x] Create a users table with fields: user_id, name, email, password (hashed), role (customer/admin).

**Server-Side Logic**

Create Express.js routes for:

- [x] Signup: Save user data in the database.

- [x] Login: Validate user credentials.

- [x] Logout: Clear session.

**Frontend Development**

- [x] Design PUG templates for signup, login, and logout pages.

- [x] Create form validation for both signup and login.

**Password Recovery (Optional)**

- [ ] Add "Forgot Password" functionality using email-based OTP.

- [ ] Update user passwords securely.

**Testing**

- [x] Test signup and login workflows.

- [x] Ensure secure session handling and logout functionality.

**Comment by Forsm0** [comment]
@ShajiRahman Any update here?

**Comment by ShajiRahman** [comment]
Yes completed the initial draft. Some changes need to made in the layout of login and register and its not working dynamically yet.

**Comment by Forsm0** [comment]
I can't see the changes; please advise where we can review them.

**Comment by ammadkhan34** [comment]
@ShajiRahman It looks good

**Comment by Forsm0** [comment]
The roles will not be implemented. Other than that it seem to be working. Please test
